addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -113,26 +122,64 @@ class EnvironmentScCidrService extends Service {
           'The Security Group for this workspace does not contain any ingress rules configured in the Service Catalog product template',
           true,
         );
-
       // Perform the actual security group ingress rule updates
       const newCidrList = await this.getUpdatedIngressRules(requestContext, {
         existingEnvironment,
         securityGroupId,
         currentIngressRules,
         updateRequest,
       });
-
       // Not storing the changes in the DB, but since all went well
       // we return the cidr field as part of the env obj
       existingEnvironment.cidr = newCidrList;
+
+      // As we removed the 443 Port CIDRs values for Rstudio during the ingress rule updates.
+      // So once it is done we replace the original values to the existing env.
+      if (productName === 'RStudioV2') {
+        cloneUpdateRequest = JSON.parse(cloneUpdateRequest);
+        existingEnvironment.cidr = cloneUpdateRequest;
+      }
     });
 
     // Write audit event
     await this.audit(requestContext, { action: 'update-environment-sc-cidr', body: existingEnvironment });
-
     return existingEnvironment;
   }
 
+  async modifyELBRule(existingEnvironment, updateRequest, albService, requestContext) {
+    // clone the update request for future use to replace the existing env data
+    const cloneUpdateRequest = JSON.stringify(updateRequest);
+    const eEnvOutputs = existingEnvironment.outputs;
+    let productName = null;
+    const metaConnection1Type = eEnvOutputs.find(obj => obj.OutputKey === 'MetaConnection1Type');
+    if (metaConnection1Type) {
+      productName = metaConnection1Type.OutputValue;
+      const listenerRuleARN = eEnvOutputs.find(obj => obj.OutputKey === 'ListenerRuleARN');
+      if (productName === 'RStudioV2' && listenerRuleARN) {
+        const cidrObj = updateRequest.find(obj => obj.fromPort === 443);
+        const ruleARN = listenerRuleARN.OutputValue;
+        const projectId = existingEnvironment.projectId;
+        const resolvedVars = {
+          prefix: 'rstudio',
+          ruleARN,
+          projectId,
+          cidr: cidrObj.cidrBlocks,
+          envId: existingEnvironment.id,
+        };
+        await albService.modifyRule(requestContext, resolvedVars);
+        // Removed the original value in 443 Port CIDRs and replace the default values in the updateRequest
+        // because the new RStudio ALB we are not storing 443 Port CIDRs into security group.
+        updateRequest.map(obj => {
+          if (obj.fromPort === 443) {
+            obj.cidrBlocks = ['0.0.0.0/0'];
+          }
+          return obj;
+        });
```

Comment:
Didn't quite catch this part. Where do we use this updateRequest? It doesn't seem like we return it. And why `0.0.0.0`?